lib/github/merge/api.ex

```diff
@@ -0,0 +1,144 @@
+defmodule BorsNG.GitHub.Merge.API do
+  @moduledoc """
+  Merge together batch patches using GitHub's API. This is
+  the default.
+  """
+
+  alias BorsNG.Worker.Batcher
+  alias BorsNG.Database.Repo
+  alias BorsNG.Database.Project
+  alias BorsNG.GitHub
+  require Logger
+
+  def merge_batch!(batch, patch_links) do
+    project = batch.project
+    repo_conn = Project.installation_connection(project.repo_xref, Repo)
+
+    stmp = "#{project.staging_branch}.tmp"
+
+    base =
+      GitHub.get_branch!(
+        repo_conn,
+        batch.into_branch
+      )
+
+    tbase = %{
+      tree: base.tree,
+      commit:
+        GitHub.synthesize_commit!(
+          repo_conn,
+          %{
+            branch: stmp,
+            tree: base.tree,
+            parents: [base.commit],
+            commit_message: "[ci skip][skip ci][skip netlify]",
```

Comment:
This is intending to skip CI when the commit is merged into the branch? Did you test this to make sure it works?

Reply:
yes, i tested it manually. i'm still writing the more automated tests

this module is copied straight from master bors; you can see this code already existed in `batcher.ex`, i just moved it out. but for why Bors is doing this in the first place, note that this function (`merge_batch!`) merges into *staging.tmp*, not staging proper, so CI is irrelevant on this branch. all the commit construction happening here is a bit subtle, because the commits created in `merge_batch!()` are never actually visible! at the bottom of `start_waiting_merged_batch()` in `batcher.ex` is where staging proper actually gets updated; you'll see that Bors completely ignores any commits created by this function and constructs a new one with the correct commit message, parented onto the batch's `into_branch` (dev in our case)